### PR TITLE
octave: use delfun of a legend to infer if hasLegend - resubmit

### LINKED ISSUE #607: Octave legend incorrectly detects all entries
```
h = plot(magic(3));
legend(h(2), 'straight line')
```
capture

### PR DESCRIPTION
**Pull request by okomarov**
Fixes #607 (Octave legend incorrectly detects all entries)

### CODE REVIEW

**Comment by PeterPablo** [comment]
I'm on the chat, in case you want to discuss something. GitHub can't merge, indicating this is not based on tip of `develop`. Try creating a new branch and cherrypicking your commits.

**Comment by okomarov** [comment]
Why is travis not picking up the last commit?

**Comment by PeterPablo** [comment]
Now it did. Did you do something or did it simply take Travis some time to notice?

**Comment by okomarov** [comment]
FFinally!

**Comment by PeterPablo** [comment]
Octave runs fine, but the output of `testGraphical` might have regressed at some point?
What do you get for `ACID(86) alphaImage`?
image

`zoom` on the other hand looks good!
image

I admit that I also didn't always check the graphical results from the testsuite (`testGraphical()`).

@okomarov -- did you verify the visual output of the tests (with changed hashes)?

**Comment by PeterPablo** [comment]
Would you please adapt `ACID(68)` to cover this new behavior? (https://github.com/matlab2tikz/matlab2tikz/issues/607#issuecomment-91139450)

**Comment by okomarov** [comment]
Octave failed on ACID(86) since the beginning. I am adapting ACID(19) not 68.

**Comment by PeterPablo** [comment]
Ok, I didn't notice that, when it was implemented. Thanks for adapting a testcase.

**Comment by PeterPablo** [comment]
**octave:**
image
Doesn't look right.

Will edit shortly for HG2.

**Comment by PeterPablo** [comment]
Ok, I guess **HG2** will be fixed by #610?!

**Comment by okomarov** [comment]
@PeterPablo Yes.

**Comment by PeterPablo** [comment]
Good! Do you agree, that the octave output needs to be fixed?

**Comment by okomarov** [comment]
Yes, I have already a fix, but it's at home...forgot to push it to my repo.

In general, the whole legend detection approach has to be re-restructured a bit, and this is part of the TODO's that Egon listed. 

In this PR I am gonna fix the problem with Octave legend alone.

**Comment by PeterPablo** [comment]
Fine by me. Please not that you will have to merge `develop` in all your PR again, since my other commit got merged. This should hopefully happen less often in the future.

**Comment by okomarov** [comment]
I know, no problem.

**Comment by PeterPablo** [comment]
Could you please merge `develop` and update the hashes once you find time. You also mentioned that you had a commit you wanted to push. Apart from that, this should be good to go?!

File: test/suites/ACID.m [hunk 1]
```diff
@@ -530,12 +530,11 @@
   stat.description = 'Test inserting of legends.';
```
Comment (PeterPablo) [inline test/suites/ACID.m]: Could you please add `stat.issue`?

**Comment by PeterPablo** [comment]
@okomarov, I locally merged `upstream` but am now wondering what your fix commit contains that you wanted to push. Currently at least `ACID(19)` is not correct (octave only), since the legend is now completely ommitted.

**Comment by okomarov** [comment]
Did not have the time to work on it and I won't be around until mid June.
In the meantime wish everyone good time.

Sent from my iPhone

On 11 May 2015, at 13:34, "Peter Pablo" <notifications@github.com<mailto:notifications@github.com>> wrote:


@okomarov<https://github.com/okomarov>, I locally merged upstream but am now wondering what your fix commit contains that you wanted to push. Currently at least ACID(19) is not correct (octave only), since the legend is now completely ommitted.

**Comment by PeterPablo** [comment]
My finding: `m2t.gcaAssociatedLegend` erroneously is empty, when `legend boxoff` is issued. As such `getAssociatedLegend()` needs to be fixed.

**Comment by okomarov** [comment]
If I recall off the top of my mind, it was using is empty instead of isAxisEmpty

Sent from my iPhone

On 11 May 2015, at 14:03, "Peter Pablo" <notifications@github.com<mailto:notifications@github.com>> wrote:


My finding: m2t.gcaAssociatedLegend erroneously is empty, when legend boxoff is issued. As such getAssociatedLegend() needs to be fixed.

**Comment by PeterPablo** [comment]
I determined this fix:
```
diff --git a/src/matlab2tikz.m b/src/matlab2tikz.m
index 7dfe975..b2e6931 100644
--- a/src/matlab2tikz.m
+++ b/src/matlab2tikz.m
@@ -956,7 +956,7 @@ function legendhandle = getAssociatedLegend(m2t, handle)
             % Make sure that m2t.legendHandles is a row vector.
             for lhandle = m2t.legendHandles(:)'
                 ud = get(lhandle, 'UserData');
-                if isVisible(lhandle) && any(handle == ud.handle)
+                if any(handle == ud.handle)
                     legendhandle = lhandle;
                     break;
                 end
```

Will issue a PR shortly.

**Comment by okomarov** [comment]
Yeah, should use isAxisVidible :+1:

Sent from my iPhone

On 11 May 2015, at 14:15, "Peter Pablo" <notifications@github.com<mailto:notifications@github.com>> wrote:


I determined this fix:

diff --git a/src/matlab2tikz.m b/src/matlab2tikz.m
index 7dfe975..b2e6931 100644
--- a/src/matlab2tikz.m
+++ b/src/matlab2tikz.m
@@ -956,7 +956,7 @@ function legendhandle = getAssociatedLegend(m2t, handle)
             % Make sure that m2t.legendHandles is a row vector.
             for lhandle = m2t.legendHandles(:)'
                 ud = get(lhandle, 'UserData');
-                if isVisible(lhandle) && any(handle == ud.handle)
+                if any(handle == ud.handle)
                     legendhandle = lhandle;
                     break;
                 end


Will issue a PR shortly.

**Comment by PeterPablo** [comment]
@okomarov, please see #653 and comment if you find the time. Thank you very much! I close here.

**Pull request closed without merging**